### PR TITLE
Contrib guide en

### PR DESCRIPTION
**Pull request by leeluolee**

### CODE REVIEW

File: en/README.md [hunk 1]
```diff
@@ -2,13 +2,14 @@
 
 svrx(server-x) is a pluggable platform built for efficient front-end development.
 
+![](https://p1.music.126.net/-ZK3lYZLWMHfPkXi7nn_gg==/109951164332910231.gif)
```
Comment (int64ago) [inline en/README.md]: - Open Chrome in `incognito mode` for privacy
 - Set VSCode language to English

File: en/README.md [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
+
 As a front-end developer, to meet different kind of development requirements, usually we will have one or more set of fixed development environment,
 in which may include a local dev server and many other debug tools.
 It's difficult to maintain a development environment: you need to install and configure every tool separately.
 Besides, you may also need to enable or disable a tool when switching among projects.
 
-To solve the problem, 
-we plan to integrate all the development services into a pluggable platform, 
+To solve the problem, we plan to integrate all the development services into a pluggable platform,
```
Comment (int64ago) [inline en/README.md]: `development services` -> `development services and tools`

File: en/README.md [hunk 3]
```diff
@@ -26,7 +27,7 @@ and instead of downloading many other packages, all you need to do is just insta
 
 ## Install & Usage
 
-You can read our docs [here](./guide/README.md).
+View [guide section](./guide/README.md) for suggestion.
```
Comment (int64ago) [inline en/README.md]: `for suggestion` -> `for more details`

File: en/contribute/plugin.md [hunk 4]
```diff
@@ -1,9 +1,868 @@
...
+    async onCreate({ logger, config }) {
+      logger.log(`Hello ${config.get('user')} from server`);
+    }
+  }
+};
+```
+
+Where
+
+1. `configSchema` Plugin param definition based json-schema ，checkout [json-schema](#schema) for more detail.
```
Comment (int64ago) [inline en/contribute/plugin.md]: `based json-schema` -> `based on JSON Schema`

File: en/contribute/plugin.md [hunk 5]
```diff
@@ -1,9 +1,868 @@
...
+   Here we just set a `user` field, it is a string
+2. `assets`: client resource setting，they will be automatically injected into the page
+
+   - `style`: css resource injection
+   - `script`: script resource injection
+
+     > All resources will be merged into one link
```
Comment (int64ago) [inline en/contribute/plugin.md]: `a single file` is better than `one link` ?

File: en/contribute/plugin.md [hunk 6]
```diff
@@ -1,9 +1,868 @@
...
+
+3. `hook.onCreate`: a function invoke when plugin been creating, we can control it by injected service
```
Comment (int64ago) [inline en/contribute/plugin.md:82]: `a function invoke when plugin been creating, we can control it by injected service`

->

`a function invoked during plugin initialization, we can overwrite it by injected services`

File: en/contribute/plugin.md [hunk 7]
```diff
@@ -1,9 +1,868 @@
...
+   in this example, we only use two service
```
Comment (int64ago) [inline en/contribute/plugin.md]: `two service` -> `two services`

File: en/contribute/plugin.md [hunk 8]
```diff
@@ -1,9 +1,868 @@
...
+
+   - `logger`: logger service
+   - `config`: config service
+
+Check out setion [service](#server) to find other services that available in `hook.onCreate`
+
+#### - `client.js`
+
+There is a global variable named `svrx` wiil be injected to every page,
```
Comment (int64ago) [inline en/contribute/plugin.md]: `wiil be injected to every page` -> `will be injected into all pages`

File: en/contribute/plugin.md [hunk 9]
```diff
@@ -1,9 +1,868 @@
...
+
+You will see `Hello svrx from browser` in console panel
+
+Unlike in server side, config in client is passed through websocket, so api is async, and return a promise
+
+> svrx also provide other client services, please check out [client api](#client) for help
+
+### publish && running
+
+When try to publish package by `npm publish`， we will find it fails. beacuse `svrx-plugin-hello-world` has beed published by [offical team](https://github.com/svrxjs/svrx-plugin-hello-world)
```
Comment (int64ago) [inline en/contribute/plugin.md]: `When try to publish package by `npm publish`， we will find it fails.`

->

`There'll be a failure when trying to publish ...`

File: en/contribute/plugin.md [hunk 10]
```diff
@@ -1,9 +1,868 @@
...
+
+So we skip this step and try the plugin directly
+
+```js
+svrx -p hello-world
+```
+
+check the terminal and browser console log, we will find `Hello svrx from browser`.
```
Comment (int64ago) [inline en/contribute/plugin.md]: `check` -> `Check`

File: en/contribute/plugin.md [hunk 11]
```diff
@@ -1,9 +1,868 @@
...
+which means plugin has worked successfully.
+
+You can also run it by programmatical way
```
Comment (int64ago) [inline en/contribute/plugin.md]: `by` -> `in`

**Comment by int64ago** [comment]
Review is still in process...

File: en/contribute/plugin.md (hunk 9 above)
Comment (leeluolee) [reply en/contribute/plugin.md]: very good.

File: en/README.md (hunk 1 above)
Comment (leeluolee) [reply en/README.md]: approve

File: en/README.md (hunk 3 above)
Comment (leeluolee) [reply en/README.md]: fixed

File: en/contribute/plugin.md (hunk 4 above)
Comment (leeluolee) [reply en/contribute/plugin.md]: fixed

File: en/contribute/plugin.md (hunk 5 above)
Comment (leeluolee) [reply en/contribute/plugin.md]: agree

File: en/contribute/plugin.md (hunk 8 above)
Comment (leeluolee) [reply en/contribute/plugin.md]: fixed

File: en/contribute/plugin.md [hunk 12]
```diff
@@ -1,9 +1,868 @@
...
+config.set('a.b', 1).then(() => {
+  config.get('a.b').then(ab => {
+    console.log(ab); // => 1
+  });
+});
+```
+
+### config schema {#schema}
+
+参数定义
```
Comment (int64ago) [inline en/contribute/plugin.md]: ?

File: en/contribute/plugin.md [hunk 13]
```diff
@@ -1,9 +1,868 @@
...
+   in this example, we only use two services
+
+   - `logger`: logger service
+   - `config`: config service
+
+Check out setion [service](#server) to find other services that available in `hook.onCreate`
+
+#### - `client.js`
+
+There is a global variable named `svrx` wiil be injected into all pages,
```
Comment (int64ago) [inline en/contribute/plugin.md]: wiil -> will

File: en/contribute/plugin.md [hunk 14]
```diff
@@ -1,9 +1,868 @@
...
+- `file:change`: triggered when any file changes
+- `ready`: triggered when server starts, if you need to handle logic after server startup (such as getting the server port), you can register this event
+
+### config
+
+config service used to modify or query the options passed by user;
+
+#### - `config.get(path)`
+
+获取一个插件配置
```
Comment (int64ago) [inline en/contribute/plugin.md]: ?

File: en/contribute/plugin.md [hunk 15]
```diff
@@ -1,9 +1,868 @@
...
+  "engines": {
+    "svrx" : "0.0.x"
+  }
+}
+```
+
+Only two fields are required by `package.json`
+
+- `name`: package name must be a string beginning with `svrx-plugin`, to help `svrx` find it in npm.
+- `engines.svrx`: Define the runnable svrx version of this plugin，svrx will automatically load the latest matching plugin
```
Comment (dayjoyx) [inline en/contribute/plugin.md:44]: > svrx will automatically load the latest matching plugin

missing a comma maybe？

File: en/contribute/plugin.md [hunk 16]
```diff
@@ -1,9 +1,868 @@
...
+
+     > All resources will be merged into a single file.
+
+3. `hook.onCreate`: a function invoke when plugin been creating, we can control it by injected service
+   in this example, we only use two services
+
+   - `logger`: logger service
+   - `config`: config service
+
+Check out setion [service](#server) to find other services that available in `hook.onCreate`
```
Comment (dayjoyx) [inline en/contribute/plugin.md]: to find other services that available in `hook.onCreate`

-> 

to find out other services available in `hook.onCreate`

File: en/contribute/plugin.md [hunk 17]
```diff
@@ -1,9 +1,868 @@
...
+
+You will see `Hello svrx from browser` in console panel
+
+Unlike in server side, config in client is passed through websocket, so api is async, and return a promise
+
+> svrx also provide other client services, please check out [client api](#client) for help
+
+### publish && running
+
+There'll be a failure when trying to publish it by `npm publish`. beacuse `svrx-plugin-hello-world` has beed published by [offical team](https://github.com/svrxjs/svrx-plugin-hello-world)
```
Comment (dayjoyx) [inline en/contribute/plugin.md]: beacuse `svrx-plugin-hello-world` has beed published

->

Beacuse `svrx-plugin-hello-world` has beed published

File: en/contribute/plugin.md [hunk 18]
```diff
@@ -1,9 +1,868 @@
...
+
+So we skip this step and try the plugin directly
+
+```js
+svrx -p hello-world
+```
+
+Check the terminal and browser console log, we will find `Hello svrx from browser`.
```
Comment (dayjoyx) [inline en/contribute/plugin.md]: maybe a comma in the end of line?

File: en/contribute/plugin.md [hunk 19]
```diff
@@ -1,9 +1,868 @@
...
+`injector` is used for rewriting the response and inject client resources.
+
+#### - `injector.add(type, resource)`
+
+Add a resource for injection , only js and css has been supported.
+
+The rule as follow.
+
+- The style will be merged into `/svrx/svrx-client.css` and
+  been injected before the closing 'head` tag
```
Comment (dayjoyx) [inline en/contribute/plugin.md]: ~~been~~ injected before the closing `head` tag

Comment (dayjoyx) [reply en/contribute/plugin.md]: and don't forget the  'head`

File: en/contribute/plugin.md [hunk 20]
```diff
@@ -1,9 +1,868 @@
...
+- The script will be merged into `/svrx/svrx-client.js` and
+  been injected before the closing 'body` tag
```
Comment (dayjoyx) [inline en/contribute/plugin.md]: same as above

File: en/contribute/plugin.md [hunk 21]
```diff
@@ -1,9 +1,868 @@
...
+`Promise`
+
+#### - `events.off(name, handler)`
+
+Remove event watcher
+
+**Usage**
+
+```js
+events.on('hello'); // remove all hello's handler
```
Comment (dayjoyx) [inline en/contribute/plugin.md]: events.off

File: en/contribute/plugin.md [hunk 22]
```diff
@@ -1,9 +1,868 @@
...
+events.on('hello', handler); // remove specific handler
```
Comment (dayjoyx) [inline en/contribute/plugin.md]: events.off

File: en/contribute/plugin.md [hunk 23]
```diff
@@ -1,9 +1,868 @@
...
+```
+
+**Param**
+
+- register({...methods}):
+  - methods: corresponding to [HTTP methods](../guide/route.md#method)
+
+#### - `router.action(name, builder)`
+
+Register a action , like [`proxy`](../guide/route.md#proxy) or [`json`](../guide/route.md#json)
```
Comment (dayjoyx) [inline en/contribute/plugin.md]: a action  -> an action

File: en/contribute/plugin.md [hunk 24]
```diff
@@ -1,9 +1,868 @@
...
+logger.notify('notify'); // show `nofity`
+logger.error('error'); // show `error` and `notify`
+logger.warn('warn'); // show `warn`、`error` and `norify`
+```
+
+> `logger.log` is an alias for `logger.notify`
+
+### io
+
+io used for the communication between server and client ， Please check it with [client-side io](#client-io)
```
Comment (dayjoyx) [inline en/contribute/plugin.md]: io is used for the communication between server and client. Please check it out in [client-side io](#client-io)

File: en/contribute/plugin.md [hunk 25]
```diff
@@ -1,9 +1,868 @@
...
+**Param**
+
+- `type`: message type
+- `payload`: message data
+
+> message payload must be serializable because they are transmitted over the network
+
+#### - `io.off(type[, handler])`
+
+解除监听
```
Comment (dayjoyx) [inline en/contribute/plugin.md]: no Chinese

File: en/contribute/plugin.md [hunk 26]
```diff
@@ -1,9 +1,868 @@
...
+          "set log level, predefined values: 'silent','notify','error','warn', 'debug'"
+      }
+    }
+  }
+};
+```
+
+**Field Details**
+
+- `type` \[String]: [JSON-Schema field type](https://json-schema.org/understanding-json-schema/reference/type.html) ，可以是`array`、`string`、`number`、`boolean`、`object`、`null`
```
Comment (dayjoyx) [inline en/contribute/plugin.md]: `可以是 `

File: zh/contribute/plugin.md [hunk 27]
```diff
@@ -431,12 +445,15 @@ route({all, get, post}=>{
 
 **Param**
 
+- register(...methods): 注册回调
+=======
```
Comment (dayjoyx) [inline zh/contribute/plugin.md:449]: these 2 lines should be removed?

**Review by dayjoyx — CHANGES_REQUESTED** [review]

File: en/contribute/plugin.md (hunk 17 above)
Comment (dayjoyx) [reply en/contribute/plugin.md]: has been

File: en/contribute/plugin.md (hunk 12 above)
Comment (leeluolee) [reply en/contribute/plugin.md]: already removed all `[ \u4e00-\u9fa5]`

File: en/contribute/plugin.md (hunk 13 above)
Comment (leeluolee) [reply en/contribute/plugin.md]: fixed

**Review by int64ago — APPROVED** [review]

**Review by dayjoyx — APPROVED** [review]

**Pull request merged**